src/Map.ts

```diff
@@ -0,0 +1,160 @@
+import { hasClass } from './decorators';
+import { is } from './object';
+import global from './global';
+
+export default class Map<K, V> {
+	private _keys: K[] = [];
+	private _values: V[] = [];
+
+	/*
+	 * An alternative to Array.prototype.indexOf using Object.is
+	 * to check for equality. See http://mzl.la/1zuKO2V
+	 */
+	private _indexOfKey(keys: K[], key: K): number {
+		for (var i = 0; i < keys.length; i++) {
+			if (is(keys[i], key)) {
+				return i;
+			}
+		}
+		return -1;
+	}
+
+	/**
+	 * Creates a new Map
+	 *
+	 * @constructor
+	 *
+	 * @param iterable
+	 * An array of two-item tuples used to initially
+	 * populate the map. The first item in each tuple
+	 * corresponds to the key of the map entry. The second
+	 * item corresponds to the value of the map entry.
+	 */
+	constructor(iterable?: any) {
+		if (!(this instanceof Map)) {
```

Comment:
If the TS Playground is any indication, TS won't allow you to directly call a class constructor, so this code would be unreachable from TS-compiled code anyway.